Training/interface/DataLoader_main.h

```diff
@@ -0,0 +1,931 @@
+               hist_weights.at(type_id)->GetYaxis()->FindBin(pt));
+
+      }
+
+      template<typename Scalar>
+      static Scalar DeltaPhi(Scalar phi1, Scalar phi2)
+      {
+          static constexpr Scalar pi = boost::math::constants::pi<Scalar>();
+          Scalar dphi = phi1 - phi2;
+          if(dphi > pi)
+              dphi -= 2*pi;
+          else if(dphi <= -pi)
+              dphi += 2*pi;
+          return dphi;
+      }
+
+
+      void FillTauBranches(const Tau& tau, Long64_t tau_i, std::shared_ptr<Data>& data)
+      {
+        Long64_t start_array_index = tau_i * n_fe_tau;
+
+        // Filling Tau Branche
+        float foo;
+        auto get_tau_branch = [&](TauFlat_f _fe) -> float& {
+            if(static_cast<int>(_fe) < 0) return foo;
+            size_t _fe_ind = static_cast<size_t>(_fe);
+            size_t index = start_array_index + _fe_ind;
+            return data->x_tau.at(index);
+        };
+
+        get_tau_branch(TauFlat_f::tau_pt) = tau.tau_pt;
+        get_tau_branch(TauFlat_f::tau_eta) = tau.tau_eta;
+        get_tau_branch(TauFlat_f::tau_phi) = tau.tau_phi;
+        get_tau_branch(TauFlat_f::tau_mass) = tau.tau_mass;
+
+        const analysis::LorentzVectorM tau_p4(tau.tau_pt, tau.tau_eta, tau.tau_phi, tau.tau_mass);
+        get_tau_branch(TauFlat_f::tau_E_over_pt) = tau_p4.energy() / tau.tau_pt;
+        get_tau_branch(TauFlat_f::tau_charge) = tau.tau_charge;
+        get_tau_branch(TauFlat_f::tau_n_charged_prongs) = tau.tau_decayMode / 5;
+        get_tau_branch(TauFlat_f::tau_n_neutral_prongs) = tau.tau_decayMode % 5;
+        get_tau_branch(TauFlat_f::chargedIsoPtSum) = tau.chargedIsoPtSum;
+        get_tau_branch(TauFlat_f::chargedIsoPtSumdR03_over_dR05) = tau.chargedIsoPtSumdR03 / tau.chargedIsoPtSum;
+        get_tau_branch(TauFlat_f::footprintCorrection) = tau.footprintCorrection;
+        get_tau_branch(TauFlat_f::neutralIsoPtSum) = tau.neutralIsoPtSum;
+        get_tau_branch(TauFlat_f::neutralIsoPtSumWeight_over_neutralIsoPtSum) = tau.neutralIsoPtSumWeight / tau.neutralIsoPtSum;
+        get_tau_branch(TauFlat_f::neutralIsoPtSumWeightdR03_over_neutralIsoPtSum) =tau.neutralIsoPtSumWeightdR03 / tau.neutralIsoPtSum;
+        get_tau_branch(TauFlat_f::neutralIsoPtSumdR03_over_dR05) = tau.neutralIsoPtSumdR03 / tau.neutralIsoPtSum;
+        get_tau_branch(TauFlat_f::photonPtSumOutsideSignalCone) = tau.photonPtSumOutsideSignalCone;
+        get_tau_branch(TauFlat_f::puCorrPtSum) = tau.puCorrPtSum;
+
+        const bool tau_dxy_valid = std::isnormal(tau.tau_dxy) && tau.tau_dxy > - 10
+                                   && std::isnormal(tau.tau_dxy_error) && tau.tau_dxy_error > 0;
+        get_tau_branch(TauFlat_f::tau_dxy_valid) = tau_dxy_valid;
+        get_tau_branch(TauFlat_f::tau_dxy) =  tau_dxy_valid ? tau.tau_dxy : 0.f;
+        get_tau_branch(TauFlat_f::tau_dxy_sig) =  tau_dxy_valid ? std::abs(tau.tau_dxy)/tau.tau_dxy_error : 0.f;
```

Comment:
Use a single if here and in all similar cases below:
```suggestion
if(tau_dxy_valid) {
    get_tau_branch(TauFlat_f::tau_dxy) = tau.tau_dxy;
    get_tau_branch(TauFlat_f::tau_dxy_sig) = std::abs(tau.tau_dxy)/tau.tau_dxy_error;
}
        
        
```